### PR TITLE
Add a tutorial notebook for Blackjack-v1

### PR DESCRIPTION
**Pull request by till2**
This PR adds the Blackjack-v1 tutorial for gym version v26.

The notebook should be compatible with Sphinx according to the [Sphinx-Notebook docs](https://sphinx-gallery.github.io/stable/tutorials/plot_parse.html#sphx-glr-tutorials-plot-parse-py).

If any changes to the tutorial are needed please let me know.

### CODE REVIEW

**Comment by pseudo-rnd-thoughts** [comment]
Currently, we only support python files not jupyter notebook though we plan to add this. You can possibly use this example https://gist.github.com/chsasank/7218ca16f8d022e02a9c0deb94a310fe

**Comment by till2** [comment]
Converted it to a python file. @pseudo-rnd-thoughts

File: docs/tutorials/blackjack_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
+"""
+Solving Blackjack with Q-Learning
+=================================
+
+"""
+
+
+######################################################################
```
Comment (mgoulao) [inline docs/tutorials/blackjack_tutorial.py:8]: This is correct but to keep some consistency change to: `# %%`.

File: docs/tutorials/blackjack_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+# Other versions of the game can be found below for you to experiment.
+
+env = gym.make('Blackjack-v1', sab=True)
+
+
+######################################################################
+# .. code:: py
+# 
+#    # Other possible environment configurations:
+#
```
Comment (mgoulao) [inline docs/tutorials/blackjack_tutorial.py:60]:
```
# %%
# .. code:: py
# 
#    # Other possible environment configurations:
# 
#    env = gym.make('Blackjack-v1', natural=True, sab=False)``
#
#    env = gym.make('Blackjack-v1', natural=False, sab=False)``
# 
```

File: docs/tutorials/blackjack_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+# game is called an episode) at a time and then update it’s Q-values after
+# each episode. The agent will have to experience a lot of episodes to
+# explore the environment sufficiently.
+# 
+# Now we should be ready to build the training loop.
+# 
+
+# hyperparameters
+epsilon = 0.6
+n_episodes = 300_000
```
Comment (mgoulao) [inline docs/tutorials/blackjack_tutorial.py:227]: Can you optimize this value to make the script finish faster without hurting the results? We execute the tutorials when building the docs so making them fast is important.

**Comment by till2** [comment]
Resolved the formatting & it is below 100k episodes now (with still mostly decent results).
Can you test if the AE-loop picture at the top shows up (if the link works)? @mgoulao

File: docs/tutorials/blackjack_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,346 @@
+"""
+Solving Blackjack with Q-Learning
+=================================
+
+"""
+
+
+# %%
+# .. image:: https://raw.githubusercontent.com/Farama-Foundation/Gymnasium/main/docs/_static/img/tutorials/blackjack_AE_loop.jpg
```
Comment (mgoulao) [inline docs/tutorials/blackjack_tutorial.py:9]: Change to `/_static/img/tutorials/blackjack_AE_loop.jpg`

File: docs/tutorials/blackjack_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+fig2 = create_plots(value_grid, policy_grid, title='Without usable ace')
+plt.show()
+
+
+# %%
+# It's good practice to call env.close() at the end of your script,
+# so that any used ressources by the environment will be closed.
+
+env.close()
+
```
Comment (mgoulao) [inline docs/tutorials/blackjack_tutorial.py:332]: Re-add the images with the plots. I don't see any advantage in executing this tutorial during build. you can revert the hyperparameters if you want.

**Comment by pseudo-rnd-thoughts** [comment]
@till2 For the future, could you resolve comments that you have solved along with a reaction or comment just so it is easier to know the progress of the PR.

File: docs/tutorials/blackjack_tutorial.py (hunk 5 above)
Comment (till2) [reply docs/tutorials/blackjack_tutorial.py:339]: readded the plots as static images to the `/_static/img/tutorials/` folder and changed the hyperparameters back to a higher number of episodes ✅

**Comment by pseudo-rnd-thoughts** [comment]
@till2 I think this is close to merging. Could you fix the pre-commit issues, https://github.com/Farama-Foundation/Gymnasium/blob/main/CONTRIBUTING.md

**Comment by till2** [comment]
> @till2 I think this is close to merging. Could you fix the pre-commit issues, https://github.com/Farama-Foundation/Gymnasium/blob/main/CONTRIBUTING.md

Should work now, it's passing locally.

**Comment by till2** [comment]
Great, ready to merge @pseudo-rnd-thoughts

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**